stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -230,11 +169,9 @@ internal class ChannelLogic(
     private suspend fun runChannelQuery(request: WatchChannelRequest): Result<Channel> {
         val offlineChannel = runChannelQueryOffline(request)
 
-        val onlineResult =
-            ChatClient.instance().queryChannelInternal(mutableState.channelType, mutableState.channelId, request)
-                .await().also { result ->
-                    onQueryChannelResult(result, mutableState.channelType, mutableState.channelId, request)
```

Comment:
Those side effects should happen only in the listeners. Exceptions to this rule only make the code harder to understand, because we find logic in places that it was not expected to be.